server/chat-commands/core.ts

```diff
@@ -1203,13 +1203,14 @@ export const commands: ChatCommands = {
 
 	search(target, room, user, connection) {
 		if (target) {
-			if (Config.laddermodchat) {
-				const userGroup = user.group;
-				if (Config.groupsranking.indexOf(userGroup) < Config.groupsranking.indexOf(Config.laddermodchat)) {
-					const groupName = Config.groups[Config.laddermodchat].name || Config.laddermodchat;
-					this.popupReply(`On this server, you must be of rank ${groupName} or higher to search for a battle.`);
-					return false;
-				}
+			if (
+				!user.hasSysopAccess() &&
+				Config.laddermodchat &&
+				Config.groupsranking.indexOf(user.group) < Config.groupsranking.indexOf(Config.laddermodchat)
```

Comment:
How much of these comparisons can be replaced with `authAtLeast`?

Should we maybe look into consolidating all our "auth at least this high" code? iirc in addition to the function in `user-groups` we also have one in `rooms` or something.